### PR TITLE
Warn if Model(Version) config fluctuates from DB state

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I implemented a warning logic to notify users about differences in the current Model(Version) configuration vs what is registered in the DB, since it will not be updated for free. Plus added guidance about how to make it sync again using CLI.

Side change: I enabled storing of `save_models_to_registry` config in DB schema, which was missed before.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: examples/nlp-case/run.py [hunk 1]
```diff
@@ -67,7 +67,7 @@
 @click.option(
     "--no-cache",
     is_flag=True,
-    default=False,
+    default=True,
```
Comment (strickvl) [inline examples/nlp-case/run.py:70]: shouldn't this get updated through the CI?

File: src/zenml/model/model_version.py [hunk 2]
```diff
@@ -470,6 +496,25 @@ def _get_model_version(self) -> "ModelVersionResponse":
...
+        configured_tags = set(self.tags or [])
+        db_tags = {t.name for t in mv.tags}
+        if db_tags != configured_tags:
+            difference["tags added"] = list(configured_tags - db_tags)
+            difference["tags removed"] = list(db_tags - configured_tags)
+        if difference:
+            logger.warning(
+                "Provided model version configuration does not match existing model "
+                f"version `{self.name}::{self.version}` with the following "
+                f"changes: {difference}. If you want to update the model version "
```
Comment (strickvl) [inline src/zenml/model/model_version.py:514]: Will this display nicely? I guess it's just logged but wondering if we can make it a bit more readable?

File: examples/nlp-case/run.py (hunk 1 above)
Comment (avishniakov) [reply examples/nlp-case/run.py:70]: It was: https://github.com/zenml-io/zenml/pull/2144/commits/d8ce54bb4c6e224e2081c61e0bd18832ca11e470

File: src/zenml/model/model_version.py (hunk 2 above)
Comment (strickvl) [reply src/zenml/model/model_version.py:514]: Was thinking to leave a linebreak at least, but if you think it looks ok then that's good enough for me. It's logging output anyway, not our CLI output etc.

**Pull request merged**